gradoop-common/src/test/java/org/gradoop/common/model/impl/pojo/ElementTest.java

```diff
@@ -0,0 +1,143 @@
+
+    try {
+      elementMock.setProperty(null);
+      fail("Expected an NullPointerException to be thrown");
+    } catch (NullPointerException nullPointerException) {
+      assertSame(nullPointerException.getMessage(), "Property was null");
+    }
+  }
+
+  @Test
+  public void testRemoveExistentProperty() {
+    Properties properties = new Properties();
+    properties.set("someKey", "someValue");
+    GradoopId gradoopIdMock = mock(GradoopId.class);
+
+    // create element mock with property
+    Element elementMock = mock(Element.class, withSettings()
+      .useConstructor(gradoopIdMock, "someLabel", properties)
+      .defaultAnswer(CALLS_REAL_METHODS));
+
+    assertEquals(properties.get("someKey"), elementMock.removeProperty("someKey"));
+    assertFalse(elementMock.hasProperty("someKey"));
+  }
+
+  @Test
+  public void testRemovePropertyNoProperties() {
+    GradoopId gradoopIdMock = mock(GradoopId.class);
+
+    // create element mock without properties
+    Element elementMock = mock(Element.class, withSettings()
+      .useConstructor(gradoopIdMock, "someLabel", null)
+      .defaultAnswer(CALLS_REAL_METHODS));
+
+    assertNull(elementMock.removeProperty("otherKey"));
+  }
+
+  @Test(expected = NullPointerException.class)
+  public void testGetPropertyValueNull() {
+    Properties properties = new Properties();
+    properties.set("someKey", "someValue");
+    GradoopId gradoopIdMock = mock(GradoopId.class);
+
+    // create element mock with property
+    Element elementMock = mock(Element.class, withSettings()
+      .useConstructor(gradoopIdMock, "someLabel", properties)
+      .defaultAnswer(CALLS_REAL_METHODS));
+
+    elementMock.getPropertyValue(null);
+  }
+
+  @Test
+  public void testGetPropertyNoProperties() {
+    GradoopId gradoopIdMock = mock(GradoopId.class);
+
+    // create element mock without properties
+    Element elementMock = mock(Element.class, withSettings()
+      .useConstructor(gradoopIdMock, "someLabel", null)
+      .defaultAnswer(CALLS_REAL_METHODS));
+
+    assertNull(elementMock.getPropertyValue("someKey"));
+  }
+
+  @Test
+  public void testHasPropertyNoProperties() {
+    GradoopId gradoopIdMock = mock(GradoopId.class);
+
+    // create element mock without properties
+    Element elementMock = mock(Element.class, withSettings()
+      .useConstructor(gradoopIdMock, "someLabel", null)
+      .defaultAnswer(CALLS_REAL_METHODS));
+
+    assertFalse(elementMock.hasProperty("someKey"));
+  }
+
+  @Test
+  public void testGetPropertyKeysNoProperties() {
+    GradoopId gradoopIdMock = mock(GradoopId.class);
+
+    // create element mock without properties
+    Element elementMock = mock(Element.class, withSettings()
+      .useConstructor(gradoopIdMock, "someLabel", null)
+      .defaultAnswer(CALLS_REAL_METHODS));
+
+    assertNull(elementMock.getPropertyKeys());
+  }
+
+  @Test
+  public void testGetPropertyKeys() {
+    Properties properties = new Properties();
```

Comment:
I would use `Properties.create()` here, although this constructor can be used I guess.

Reply:
done